solidity/contracts/maintainer/MaintainerProxy.sol

```diff
@@ -0,0 +1,588 @@
+    /// @param walletMainUtxo Data of the wallet's main UTXO, as currently
+    ///        known on the Ethereum chain.
+    function notifyCloseableWallet(
+        bytes20 walletPubKeyHash,
+        BitcoinTx.UTXO calldata walletMainUtxo
+    ) external onlyMaintainer {
+        uint256 gasStart = gasleft();
+
+        bridge.notifyCloseableWallet(walletPubKeyHash, walletMainUtxo);
+
+        reimbursementPool.refund(
+            (gasStart - gasleft()) + _notifyCloseableWalletGasOffset,
+            msg.sender
+        );
+    }
+
+    /// @notice Notifies about the end of the closing period for the given wallet.
+    ///         Closes the wallet ultimately and notifies the ECDSA registry
+    ///         about this fact.
+    /// @param walletPubKeyHash 20-byte public key hash of the wallet
+    function notifyWalletClosingPeriodElapsed(bytes20 walletPubKeyHash)
+        external
+        onlyMaintainer
+    {
+        uint256 gasStart = gasleft();
+
+        bridge.notifyWalletClosingPeriodElapsed(walletPubKeyHash);
+
+        reimbursementPool.refund(
+            (gasStart - gasleft()) + _notifyWalletClosingPeriodElapsedGasOffset,
+            msg.sender
+        );
+    }
+
+    /// @notice Allows to defeat a pending fraud challenge against a wallet if
+    ///         the transaction that spends the UTXO follows the protocol rules.
+    ///         In order to defeat the challenge the same `walletPublicKey` and
+    ///         signature (represented by `r`, `s` and `v`) must be provided as
+    ///         were used to calculate the sighash during input signing.
+    ///         The fraud challenge defeat attempt will only succeed if the
+    ///         inputs in the preimage are considered honestly spent by the
+    ///         wallet. Therefore the transaction spending the UTXO must be
+    ///         proven in the Bridge before a challenge defeat is called.
+    ///         If successfully defeated, the fraud challenge is marked as
+    ///         resolved and the amount of ether deposited by the challenger is
+    ///         sent to the treasury.
+    /// @param walletPublicKey The public key of the wallet in the uncompressed
+    ///        and unprefixed format (64 bytes)
+    /// @param preimage The preimage which produces sighash used to generate the
+    ///        ECDSA signature that is the subject of the fraud claim. It is a
+    ///        serialized subset of the transaction. The exact subset used as
+    ///        the preimage depends on the transaction input the signature is
+    ///        produced for. See BIP-143 for reference
+    /// @param witness Flag indicating whether the preimage was produced for a
+    ///        witness input. True for witness, false for non-witness input
+    function defeatFraudChallenge(
+        bytes calldata walletPublicKey,
+        bytes calldata preimage,
+        bool witness
+    ) external onlyMaintainer {
```

Comment:
Do we really want the `defeatFraudChallenge` function to be called by maintainers only? It's mentioned in https://github.com/keep-network/tbtc-v2/issues/171 as `operator-only` type. I also remember talking with @pdyraga that calling this function is supposed to be incentivized by some compensation from treasury, not gas reimbursement.